pkg/controller/core/queuedworkload_controller.go

```diff
@@ -190,6 +190,10 @@ func (r *QueuedWorkloadReconciler) Update(e event.UpdateEvent) bool {
 		if err := r.cache.DeleteWorkload(oldWl); err != nil {
 			log.Error(err, "Failed to delete workload from cache")
 		}
+
+		// Clean the workload from the ClusterQueue if it's present and trigger
+		// the relevant inadmissibleWorkloads movement if required.
+		r.queues.DeleteWorkload(oldWl)
```

Comment:
While removing the check at line 140 above is ok, invoking this here is unexpected; I think we should have a function in queue manager that given an "event", it moves indadmissible workloads; we invoke this function explicitly here and in the Delete callback above. We can start with a simple API for this function, just explicitly pass in a `kueue.QueuedWorkload` type.

Reply:
I add a new public function in manager. Invoke it in `queuedworkload_controller.go`